File: backend/services/core-service/src/main/java/kr/lastdish/core/order/domain/OrderStatus.java
package kr.lastdish.core.order.domain;

public enum OrderStatus {
  RESERVED("예약 완료"),
  PICKUP_READY("픽업 대기"),
  PICKED_UP("픽업 완료"),
  NO_SHOW("노쇼"),
  CANCELLED("취소"),
  REJECTED("거절");

  private final String displayName;

  OrderStatus(String displayName) {
    this.displayName = displayName;
  }

  public String getDisplayName() {
    return displayName;
  }

  public boolean canTransitionTo(OrderStatus nextStatus) {
    return switch (this) {
      case RESERVED ->
          nextStatus == PICKUP_READY || nextStatus == CANCELLED || nextStatus == REJECTED;
      case PICKUP_READY -> nextStatus == PICKED_UP || nextStatus == NO_SHOW;
      case PICKED_UP, NO_SHOW, CANCELLED, REJECTED -> false;
    };
  }
}

File: frontend/lib/domain/model/order_status.dart
/// 주문 상태(백엔드 `OrderStatus` enum, ADR 014로 6개 확정). 서버는 이 값 이름
/// 그대로(영문 enum명)를 JSON으로 주고받는다 — category.dart와 같은 패턴.
const List<String> orderStatusValues = [
  'RESERVED',
  'PICKUP_READY',
  'PICKED_UP',
  'NO_SHOW',
  'CANCELLED',
  'REJECTED',
];

const Map<String, String> orderStatusDisplayNames = {
  'RESERVED': '예약',
  'PICKUP_READY': '픽업대기',
  'PICKED_UP': '픽업완료',
  'NO_SHOW': '노쇼',
  'CANCELLED': '취소',
  'REJECTED': '거절',
};

String orderStatusLabel(String value) => orderStatusDisplayNames[value] ?? value;
